### PR TITLE
systemd: more functional component porting of trivial components

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: pkg/systemd/reporting.jsx [hunk 1]
```diff
@@ -67,38 +68,25 @@ function get_problem_properties(problem) {
...
-    updateStatusFromBus() {
-        get_problem_properties(this.props.problem.path)
+    const updateStatusFromBus = () => {
+        get_problem_properties(problem.path)
                 .catch(exception => {
-                    this.setState({ problemState: ProblemState.UNREPORTABLE });
-
-                    console.error(cockpit.format("Getting properties for problem $0 failed: $1", this.props.problem.path, exception));
+                    setProblemState(ProblemState.UNREPORTABLE);
+                    console.error(cockpit.format("Getting properties for problem $0 failed: $1", problem.path, exception));
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:81]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 2]
```diff
@@ -67,38 +68,25 @@ function get_problem_properties(problem) {
...
                 })
                 .then((properties) => {
                     if (!properties) {
                         return;
                     }
 
                     if (!properties[0].CanBeReported.v) {
-                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                        setProblemState(ProblemState.UNREPORTABLE);
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:89]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 3]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
-    _onCancelButtonClick(event) {
-        this.state.process.close("canceled");
-    }
+    useInit(() => {
+        updateStatusFromBus();
+    });
 
-    _onReportButtonClick(event) {
-        this.setState({ problemState: ProblemState.UNREPORTABLE });
+    const onCancelButtonClick = _event => process.close("canceled");
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:117]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 4]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
+    const onReportButtonClick = _event => {
+        setProblemState(ProblemState.UNREPORTABLE);
+        const process = cockpit.spawn(["reporter-ureport", "-d", problem.ID],
                                       {
                                           err: "out",
                                           superuser: "true",
                                       })
-                .stream((data) => this.setState({ message: data, }))
-                .then(() => this.setState({ problemState: ProblemState.REPORTED, }))
+                .stream((data) => setMessage(data))
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:126]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 5]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
+                .then(() => setProblemState(ProblemState.REPORTED))
                 .catch(exception => {
-                    this.setState({ problemState: ProblemState.REPORTABLE, });
+                    setProblemState(ProblemState.REPORTABLE);
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:129]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 6]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
+    const [reportLinks, setReportLinks] = React.useState([]);
+    const [task, setTask] = React.useState(null);
+    const label = workflow[1];
+    const inputRef = React.createRef();
+
+    const updateStatusFromBus = () => {
+        const on_get_properties = properties => {
+            if (!properties[0].CanBeReported.v) {
+                setProblemState(ProblemState.UNREPORTABLE);
+                return;
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:162]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 7]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
-                            reportLinks={this.state.reportLinks}
-        />;
-    }
-}
+            for (const report of properties[0].Reports.v) {
+                if (!("WORKFLOW" in report[1])) {
+                    continue;
+                }
+                if (workflow[0] !== report[1].WORKFLOW.v.v) {
+                    continue;
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:173]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 8]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
                             return;
                         }
 
-                        console.error(cockpit.format("reportd task for workflow $0 did not finish: $1", this.props.workflow[0], (ex.problem || ex.message)));
-                        this.setState({ message: _("Reporting failed") });
+                        console.error(cockpit.format("reportd task for workflow $0 did not finish: $1", workflow[0], (ex.problem || ex.message)));
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:322]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 9]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
...
+                .then(() => createTask(client))
                 .catch(exception => {
-                    const message = cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message));
-
-                    this.setState({
-                        message,
-                        problemState: ProblemState.REPORTABLE,
-                    });
+                    setMessage(cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message)));
+                    setProblemState(ProblemState.REPORTABLE);
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:339]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 10]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
...
 
-    getWorkflows(client) {
-        client.call("/org/freedesktop/reportd/Service", "org.freedesktop.reportd.Service", "GetWorkflows", [this.props.problem.path])
-                .then((args, options) => this.setState({ workflows: args[0] }),
-                      exception => console.error(cockpit.format("Failed to get workflows for problem $0: $1", this.props.problem.path, (exception.problem || exception.message))));
-    }
+    const getWorkflows = client => {
+        client.call("/org/freedesktop/reportd/Service", "org.freedesktop.reportd.Service", "GetWorkflows", [problem.path])
+                .then((args, _options) => setWorkflows(args[0]),
+                      exception => console.error(cockpit.format("Failed to get workflows for problem $0: $1", problem.path, (exception.problem || exception.message))));
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:423]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 11]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
...
+    };
 
-        this.setState({
-            message: _("Waiting to start…"),
-            problemState: ProblemState.REPORTING,
-        });
+    const onReportButtonClick = (_event) => {
+        setProblemState(ProblemState.UNREPORTABLE);
+        setMessage(_("Waiting to start…"));
+        setProblemState(ProblemState.REPORTING);
```
Comment (subhoghoshX) [inline pkg/systemd/reporting.jsx:332]: This looks an issue coming from the old code, but why change the `problemState` state twice in a row? I think React will batch the `setProblemState` calls and only one of them (the last one) will take effect.

File: pkg/systemd/reporting.jsx [hunk 12]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
...
                 .catch(exception => {
-                    const message = cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message));
-
-                    this.setState({
-                        message,
-                        problemState: ProblemState.REPORTABLE,
-                    });
+                    setMessage(cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message)));
+                    setProblemState(ProblemState.REPORTABLE);
                     console.error(message);
```
Comment (subhoghoshX) [inline pkg/systemd/reporting.jsx:340]: Removing the `message` variable above might break this line *subtly*. As `setMessage()` doesn't immediately change the state, the `message` state variable will still have the old value.

**Review by subhoghoshX — COMMENTED** [review]
Great effort. Thanks :)

File: pkg/systemd/reporting.jsx (hunk 11 above)
Comment (jelly) [reply pkg/systemd/reporting.jsx:332]: Yes, good point, we should just drop `UNREPORTABLE`

File: pkg/systemd/reporting.jsx (hunk 12 above)
Comment (jelly) [reply pkg/systemd/reporting.jsx]: Good catch, fixed!

File: pkg/systemd/reporting.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:81]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:89]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:117]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 4 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:126]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 5 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:129]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:162]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:173]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 13]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
-class BusWorkflowRow extends React.Component {
-    constructor(props) {
-        super(props);
+            if (reported) {
+                setProblemState(ProblemState.REPORTED);
+                setReportLinks(reportLinks);
+            }
+        };
+        const on_get_properties_rejected = exception => {
+            setProblemState(ProblemState.UNREPORTABLE);
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:191]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 14]
```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
...
 
-        this.state = {
-            label: this.props.workflow[1],
-            message: "",
-            problemState: ProblemState.REPORTABLE,
-            reportLinks: [],
-            task: null,
+            console.error(cockpit.format("Getting properties for problem $0 failed: $1", problem.path, exception));
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:193]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx (hunk 8 above)
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:322]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 15]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
...
-            }
-
-            if (reported) {
-                this.setState({
-                    problemState: ProblemState.REPORTED,
-                    reportLinks,
+                    const newMessage = cockpit.format("reportd task could not be created: $0", (exception.problem || exception.message));
+                    setMessage(newMessage);
+                    setProblemState(ProblemState.REPORTABLE);
+                    console.error(newMessage);
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:340]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/reporting.jsx [hunk 16]
```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
...
 
-    getWorkflows(client) {
-        client.call("/org/freedesktop/reportd/Service", "org.freedesktop.reportd.Service", "GetWorkflows", [this.props.problem.path])
-                .then((args, options) => this.setState({ workflows: args[0] }),
-                      exception => console.error(cockpit.format("Failed to get workflows for problem $0: $1", this.props.problem.path, (exception.problem || exception.message))));
-    }
+    const getWorkflows = client => {
+        client.call("/org/freedesktop/reportd/Service", "org.freedesktop.reportd.Service", "GetWorkflows", [problem.path])
+                .then((args, _options) => setWorkflows(args[0]),
+                      exception => console.error(cockpit.format("Failed to get workflows for problem $0: $1", problem.path, (exception.problem || exception.message))));
```
Comment (cockpituous) [inline pkg/systemd/reporting.jsx:423]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by subhoghoshX — APPROVED** [review]
Perfect!!

**Review by martinpitt — APPROVED** [review]
"don't use this.props in a functional component " is a weird description for "port ProfilesMenuDialogBody to functional component", but changes look good. Thanks!

**Pull request merged**